js/utils.js

```diff
@@ -45,6 +45,71 @@ Fliplet.Registry.set('dynamicListUtils', (function() {
     return parseFloat(value);
   }
 
+  function getFilesInfo(options) {
+    var entry = options.entryData;
+    var detailViewFileOptions = _.filter(options.detailViewOptions, function(option) {
+      return option.editable && option.type === 'file';
+    });
+
+    var formFilesInfoInDetailViewOptions = detailViewFileOptions.map(function(detailViewFileOption) {
+      return new Promise(function(resolve) {
+        var label = '';
+        var labelEnabled = true;
+        var files = typeof entry.originalData[detailViewFileOption.column] === 'string' ?
+          splitByCommas(entry.originalData[detailViewFileOption.column]) :
+          entry.originalData[detailViewFileOption.column];
+
+        if (!files) {
+          return resolve();
+        }
+
+        switch (detailViewFileOption.fieldLabel) {
+          case 'column-name':
+            if (detailViewFileOption.column !== 'custom') {
+              label = detailViewFileOption.column;
+            }
+
+            break;
+          case 'custom-label':
+            label = new Handlebars.SafeString(Handlebars.compile(detailViewFileOption.customFieldLabel)(entry.originalData));
+
+            break;
+          case 'no-label':
+            labelEnabled = false;
+
+            break;
+          default:
+            break;
+        }
+
+        var filesDetailViewInfo = Promise.all(files.map(function(fileUrl) {
+          var fileId = Fliplet.Media.isRemoteUrl(fileUrl)[2];
+
+          return Fliplet.Media.Files.get(fileId).then(function(file) {
```

Comment:
@squallstar I think we have something that could create a server load issue here.

In order to print file information on the page, LFD needs to make a request to find out file information. If there are 1000 LFD entries and each entry has 20 files attached, then this will make 20k requests at once.

I think we'll need:

1. An endpoint and JS API to get the file information for multiple files at once
2. Cache the response so that it doesn't get the file information for a file that has been loaded before (this can be done internally via the JS API or by the LFD code)

Thoughts?

Reply:
1. Agreed (around 2h work)
2. Agreed

Reply:
@squallstar Will it be possible and suitable for the JS API to handle the caching, and how long will that take?

Reply:
This is the mockup for what we're aiming to display in the LFD detail view, based on the data found in each data source entry. The data source column will likely contain a list of file entry IDs for each entry.

![image](https://user-images.githubusercontent.com/290733/105887303-763fe780-6003-11eb-9d70-d74496d9a8d0.png)

Each file will need its uploaded date and size displayed. To do so, the LFD needs to get the file metadata. And I assume once this is retrieved, there's really no reason that they would change again, hence the caching recommendation.

Reply:
@squallstar Can you confirm that you've done both of the below as previously mentioned:

1. An endpoint and JS API to get the file information for multiple files at once
2. Cache the response so that it doesn't get the file information for a file that has been loaded before (this can be done internally via the JS API or by the LFD code). Did you update the JS API to handle this?